src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -28,17 +28,37 @@ public static class GetDownloadResultUtility
            ILogger logger,
            CancellationToken token)
         {
-            // This code should respect -NoCache option and not read packages from the global
-            // packages folder: https://github.com/NuGet/Home/issues/1406
-            var packageFromGlobalPackages = GlobalPackagesFolderUtility.GetPackage(
-                identity,
-                globalPackagesFolder);
+            // Observe the NoCache argument.
+            var directDownload = downloadContext.DirectDownload;
+            DownloadResourceResult packageFromGlobalPackages = null;
+            try
+            {
+                packageFromGlobalPackages = GlobalPackagesFolderUtility.GetPackage(
+                    identity,
+                    globalPackagesFolder);
 
-            if (packageFromGlobalPackages != null)
+                if (packageFromGlobalPackages != null)
+                {
+                    if (!downloadContext.SourceCacheContext.NoCache)
+                    {
+                        return packageFromGlobalPackages;
+                    }
+                    else
+                    {
+                        // The package already exists in the global packages folder but the caller has requested NoCache,
+                        // which means the package in the global packages folder should not be used. In this particular
+                        // case, NoCache needs to imply DirectDownload.
+                        directDownload = true;
+                        packageFromGlobalPackages.Dispose();
```

Comment:
When it gets the latest content from server with NoCache argument, should it also update it in cache? My assumption is since user didn't specify the directDownload flag so it should write the latest content in cache.